src/Serilog.Sinks.OpenTelemetry/Sinks/OpenTelemetry/ProtocolHelpers/PrimitiveConversions.cs

```diff
@@ -75,26 +76,37 @@ public static KeyValue NewStringAttribute(string key, string value)
         });
     }
 
-    public static AnyValue? ToOpenTelemetryPrimitive(object? value)
+    public static AnyValue ToOpenTelemetryPrimitive(object? value)
     {
         return value switch
         {
+            null => new AnyValue(),
             short i => new AnyValue { IntValue = i },
             int i => new AnyValue { IntValue = i },
             long i => new AnyValue { IntValue = i },
+            byte i => new AnyValue { IntValue = i },
             ushort i => new AnyValue { IntValue = i },
             uint i => new AnyValue { IntValue = i },
-            ulong i => new AnyValue { IntValue = (long)i },
+            ulong i and < long.MaxValue => new AnyValue { IntValue = (long)i },
+#if FEATURE_HALF
+            Half d => new AnyValue { DoubleValue = (double)d },
```

Comment:
More types need to be filled in, we can use those handled by Serilog's `JsonValueFormatter` as a starting point.